client/state/shortcodes/reducer.js

```diff
@@ -0,0 +1,69 @@
+/**
+ * External dependencies
+ */
+import { combineReducers } from 'redux';
+import { has } from 'lodash';
+
+/**
+ * Internal dependencies
+ */
+import { shortcodesSchema } from './schema';
+import { createReducer } from 'state/utils';
+import {
+	SHORTCODE_RECEIVE,
+	SHORTCODE_REQUEST,
+	SHORTCODE_REQUEST_FAILURE,
+	SHORTCODE_REQUEST_SUCCESS
+} from 'state/action-types';
+
+const requestingShortcode = ( state, siteId, shortcode, requesting ) => {
+	const siteShortcodes = has( state, siteId ) ? state[ siteId ] : {};
+	return {
```

Comment:
`merge()` works great for that purpose indeed, thanks (we're not unsetting anyway)! And `...state[ siteId ]` also works fine, as long as state is not `null` for some reason (which it should not be). I've updated the PR with both of these, it looks very simplified now.